rmanager.py

```diff
@@ -0,0 +1,46 @@
+class ResourceManager(object):
+    def __init__(self, ESstats, cpu_percent_usage, memory_percent_usage, njobs):
+        self.machineResources = { "total": {"cpu_75": cpu_percent_usage*int(ESstats["MachineCPUCount"]), 
+                                            "rss_75" : memory_percent_usage*int(ESstats["MachineMemoryGB"])*10737418 },
+                                  "available": {"cpu_75": cpu_percent_usage*int(ESstats["MachineCPUCount"]),
+                                            "rss_75" : memory_percent_usage*int(ESstats["MachineMemoryGB"])*10737418 } }
+        self.externalsStats = ESstats["externals"]# this should be get before running pkgtools
+        self.resourcesAllocatedForExternals = {} # resources that were alocated for given externals, say root -> "root": {"rss":"", cpu:""}
+        self.jobsOrderMetric = "cpu_75" # default to cpu
```

Comment:
may be using total build time as ordering matrix will help. For example we want to run jobs which takes more time first so that we do not have a long tail